src/IssueViz.UnitTests/Editor/LocationTagging/LocationTaggerTests.cs

```diff
@@ -0,0 +1,431 @@
+        }
+
+        [TestMethod]
+        public void BufferChanged_SnapshotsAreTranslated()
+        {
+            const int length = 100;
+            var bufferMock = CreateBufferMock(ValidBufferDocName, length);
+
+            var versionMock2 = CreateTextVersion(bufferMock.Object, 2);
+            var versionMock1 = CreateTextVersion(bufferMock.Object, 1, nextVersion: versionMock2);
+
+            var snapshotMock1 = CreateSnapshot(length, bufferMock.Object, versionMock1);
+            var snapshotMock2 = CreateSnapshot(length, bufferMock.Object, versionMock2);
+
+            var locSpan1 = new Span(1, 10);
+            var storeMock = CreateStoreWithLocationsWithSpans(snapshotMock1, locSpan1);
+
+            var testSubject = new LocationTagger(bufferMock.Object, storeMock, ValidSpanCalculator, ValidLogger);
+
+            // Sanity checks
+            testSubject.TagSpans.Count().Should().Be(1);
+            testSubject.TagSpans[0].Span.Snapshot.Should().Be(snapshotMock1);
+            testSubject.TagSpans[0].Tag.Location.Span.Value.Snapshot.Should().Be(snapshotMock1);
+
+            RaiseBufferChangedEvent(bufferMock, snapshotMock1, snapshotMock2);
+
+            testSubject.TagSpans.Count().Should().Be(1);
+            testSubject.TagSpans[0].Span.Snapshot.Should().Be(snapshotMock2);
+            testSubject.TagSpans[0].Tag.Location.Span.Value.Snapshot.Should().Be(snapshotMock2);
+        }
+
+        private static Mock<ITextBuffer> CreateBufferMock(string filePath, int length = 999)
+        {
+            var snapshotMock = new Mock<ITextSnapshot>();
+            snapshotMock.Setup(x => x.Length).Returns(length);
+
+            var bufferMock = new Mock<ITextBuffer>();
+            bufferMock.Setup(x => x.CurrentSnapshot).Returns(snapshotMock.Object);
+
+            var properties = new Microsoft.VisualStudio.Utilities.PropertyCollection();
+            bufferMock.Setup(x => x.Properties).Returns(properties);
+
+            var docMock = new Mock<ITextDocument>();
+            docMock.Setup(x => x.FilePath).Returns(filePath);
+            properties[typeof(ITextDocument)] = docMock.Object;
+
+            return bufferMock;
+        }
+
+        private static IAnalysisIssueLocationVisualization CreateLocViz(SnapshotSpan? span)
+        {
+            var locVizMock = new Mock<IAnalysisIssueLocationVisualization>();
+            locVizMock.SetupProperty(x => x.Span);
+            locVizMock.Object.Span = span;
+
+            locVizMock.Setup(x => x.Location).Returns(Mock.Of<IAnalysisIssueLocation>());
+            return locVizMock.Object;
+        }
+
+        private static int spanCounter;
+
+        private static SnapshotSpan CreateUniqueSpan()
+        {
+            // NB SnapshotSpans are value types, so any "new SpanshotSpan() == new SpanshotSpan()"
+            // If we want to differentiate between spans we can do so by creating
+            // them with different lengths.
+            var length = ++spanCounter;
```

Comment:
Cosmetics: if the length doesn't matter, we can just do Guid.NewGuid().GetHashCode() for a random int.

Reply:
To me, that seems like a pretty non-intuitive way of generating a unique int, and `GetHashCode` isn't guaranteed to be unique.